### PR TITLE
feat: add C implementation for `math/base/special/fast/pow-int`

### PR DESCRIPTION
**Pull request by gunjjoshi**
Resolves a part of #1917.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/fast/pow-int`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/fast/pow-int).

```C
double stdlib_base_fast_pow( const double x, const int32_t y )
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #1917.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   ☑️ Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/fast/pow-int/README.md [hunk 1]
```diff
@@ -138,6 +138,97 @@ for ( y = 0; y < 309; y++ ) {
...
+// returns 1.0
+```
+
+The function accepts the following arguments:
+
+-   **x**: `[in] double` base.
+-   **y**: `[in] double` exponent.
+
+```c
+double stdlib_base_fast_pow( const double x, const int32_t y );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/pow-int/README.md:185]:
```suggestion
double stdlib_base_fast_pow( const double base, const int32_t exponent );
```

File: lib/node_modules/@stdlib/math/base/special/fast/pow-int/README.md [hunk 2]
```diff
@@ -138,6 +138,97 @@ for ( y = 0; y < 309; y++ ) {
...
+
+int main( void ) {
+    const double x[] = { 3.14, 2.0, 2.0, 0.0 };
+    const int32_t y[] = { 0, 3, -2, 0 };
+
+    double z;
+    int i;
+    for ( i = 0; i < 4; i++ ) {
+        z = stdlib_base_fast_pow( x[ i ], y[ i ] );
+        printf( "pow( %lf, %lf ) = %lf\n", x[ i ], y[ i ], z );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/pow-int/README.md]:
```suggestion
        printf( "pow( %lf, %d ) = %lf\n", x[ i ], y[ i ], z );
```

File: lib/node_modules/@stdlib/math/base/special/fast/pow-int/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+int main( void ) {
+	const double x[] = { 3.14, 2.0, 2.0, 0.0 };
+	const int32_t y[] = { 0, 3, -2, 0 };
+
+	double z;
+	int i;
+	for ( i = 0; i < 4; i++ ) {
+		z = stdlib_base_fast_pow( x[ i ], y[ i ] );
+		printf( "pow( %lf, %lf ) = %lf\n", x[ i ], y[ i ], z );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/pow-int/examples/c/example.c]:
```suggestion
		printf( "pow( %lf, %d ) = %lf\n", x[ i ], y[ i ], z );
```

File: lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+* @return     function value
+*
+* @example
+* double y = stdlib_base_fast_pow( 2.0, 3 );
+* // returns 8.0
+*/
+double stdlib_base_fast_pow( const double x, const int32_t y ) {
+	double v;
+	double xx;
+	int32_t yy;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c:40]:
```suggestion
	double xc;
	int32_t yc;
```

As per conventions

File: lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+	xx = x;
+	yy = y;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c]:
```suggestion
	xc = x;
	yc = y;
```

File: lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	if ( stdlib_base_is_nan( x ) ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c]:
```suggestion
	if ( stdlib_base_is_nan( xc ) ) {
```

File: lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+		yy = -yy;
+		if ( x == 0.0 ) {
+			xx = 1.0 / x;
+			if ( ( yy & ONE ) == ONE ) {
+				return xx;
+			}
+			return STDLIB_CONSTANT_FLOAT64_PINF;
+		}
+		xx = 1.0 / x;
+	}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c]:
```suggestion
	if ( yc < ZERO ) {
		yc = -yc;
		if ( xc == 0.0 ) {
			xc = 1.0 / xc;
			if ( ( yc & ONE ) == ONE ) {
				return xc;
			}
			return STDLIB_CONSTANT_FLOAT64_PINF;
		}
		xc = 1.0 / xc;
	}
```

File: lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+	v = 1.0;
+	while ( yy != ZERO ) {
+		if ( ( yy & ONE ) == ONE ) {
+			v *= xx;
+		}
+		xx *= xx;
+		yy >>= ONE;
+	}
+	return v;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c]:
```suggestion
	else if ( yc == ZERO ) {
		return 1.0;
	}
	v = 1.0;
	while ( yc != ZERO ) {
		if ( ( yc & ONE ) == ONE ) {
			v *= xc;
		}
		xc *= xc;
		yc >>= ONE;
	}
	return v;
```

File: lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c (hunk 4 above)
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/pow-int/src/main.c]:
```suggestion
	int32_t yc;
	double xc;
	double v;
```

Order by length and as per conventions `xc`, `yc`

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @gunjjoshi !

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**